Document custom resource encryption support (#38109)
kubernetes/website: content/en/docs/tasks/administer-cluster/securing-a-cluster.md

- liggitt
- ericchiang

> - destijl

Reviewer: Remove destijl and ericchiang, they are no longer involved in upstream Kubernetes.

your backups using a well reviewed backup and encryption solution, and consider using full disk
encryption where possible.

> As a beta feature, Kubernetes supports [encryption at rest](/docs/tasks/administer-cluster/encrypt-data/).

Reviewer 2: Only the v1 API is in beta, so should we call those out here? v1 - beta and v2 - alpha?

Reviewer 3: Encryption at rest has been beta since Kubernetes v1.13, unless it graduated since then and the docs are stale.

The v1 config API is beta with an a v2 config API in alpha. That's not the same as the feature itself. We've had a bunch of features (eg the API for Pods) where the feature was stable, but a key API that enables them to actually work (eg CRI) was still alpha.

Reviewer: Encryption at rest is GA (for secret box, aes, etc).

Encryption at rest for KMS v1 is beta.

Encryption at rest for KMS v2 is alpha.

Reviewer 3: ```suggestion
Kubernetes supports optional [encryption at rest](/docs/tasks/administer-cluster/encrypt-data/)
for information in the Kubernetes API.
```
based on https://github.com/kubernetes/website/pull/38109#discussion_r1034199324

This lets you ensure that when Kubernetes stores data for objects (for example, `Secret` or

> `ConfigMap`) objects, the API server writes an encrypted representation of the object.

Reviewer 2: ```suggestion
`ConfigMap`), the API server writes an encrypted representation of the object.
```

Reviewer 3: Moving the paren is better:
```suggestion
This lets you ensure that when Kubernetes stores data for objects (for example, `Secret` or
`ConfigMap` objects), the API server writes an encrypted representation of the object.
```

We like docs to be clear, even to the point of some repetition.

That encryption means that even someone who has access to etcd backup data is unable
to view the content of those objects.

> In Kubernetes {{< skew currentVersion >}} you can also can encrypt custom resources;

Reviewer 2: ```suggestion
In Kubernetes {{< skew currentVersion >}} you can also encrypt custom resources;
```

Reviewer 3: Oops!

Reviewer 3: Not that much repetition.

your backups using a well reviewed backup and encryption solution, and consider using full disk
encryption where possible.

> As a beta feature, Kubernetes supports [encryption at rest](/docs/tasks/administer-cluster/encrypt-data/). This lets you ensure that when Kubernetes stores data for objects (for example, `Secret` or `ConfigMap`) objects, the API server writes an encrypted representation of the object. That encryption means that even someone who has access to etcd backup data is unable to view the content of those objects. In Kubernetes {{< skew currentVersion >}}, you can also encrypt custom resources; encryption-at-rest for extension APIs defined in CustomResourceDefinitions was added to Kubernetes v1.26.

Reviewer 3: Please retain the wrapping. Long paragraphs all on one line are hard to localize.

encryption where possible.

> Kubernetes supports [encryption at rest](/docs/tasks/administer-cluster/encrypt-data/), a feature 
> introduced in 1.7, v1 beta since 1.13, and v2 alpha since 1.25. This will encrypt resources like `Secret` and `ConfigMap` (Custom Resources starting v1.26) in etcd, preventing
> parties that gain access to your etcd backups from viewing the content of those secret data. While
> this feature is currently beta, it offers an additional level of defense when backups
> are not encrypted or an attacker gains read access to etcd.

Reviewer 3: ```suggestion
As a beta feature, Kubernetes supports [encryption at rest](/docs/tasks/administer-cluster/encrypt-data/).
This lets you ensure that when Kubernetes stores data for objects (for example, `Secret` or
`ConfigMap`) objects, the API server writes an encrypted representation of the object.
That encryption means that even someone who has access to etcd backup data is unable
to view the content of those objects.

In Kubernetes {{< skew currentVersion >}} you can also can encrypt custom resources;
encryption-at-rest for extension APIs defined in CustomResourceDefinitions was added to
Kubernetes as part of the v1.26 release.
```
